### PR TITLE
Feature getting started guide

### PR DESCRIPTION
**Pull request by fmitchell-r7**
This walks through the creation of a VPC with public and private subnets. The VPC is built out step-by-step with Convection, and the result matches up with "Scenario 2" from Amazon's VPC Wizard Scenarios.

http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Scenario2.html

### CODE REVIEW

**Comment by athompson-r7** [comment]
This is great!

To better show the power, I would try to sneak in `attribute` and `stack.get()` or `stack[]`.  You also might split this into two stacks to show the data sharing piece.

File: guides/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,1138 @@
...
+ create  .Resources.PublicSubnet.Type: AWS::EC2::Subnet
+ create  .Resources.PublicSubnet.Properties.VpcId.Ref: DemoVPC
+ create  .Resources.PublicSubnet.Properties.CidrBlock: 10.10.11.0/24
+ create  .Resources.PublicSubnet.Properties.MapPublicIpOnLaunch: true
+ create  .Resources.PublicSubnet.Properties.Tags.0.Key: Name
+ create  .Resources.PublicSubnet.Properties.Tags.0.Value: convection-demo-vpc-public
+```
+
+Convection says it will delete the "MapPublicIpOnLaunch" from our private
+subnet. That's fine, because the subnet is private. The public subnet will be
```
Comment (dgreene-r7) [inline guides/getting-started.md]: The sentence on 401-402 would benefit from a little bit extra about why `MapPublicIpOnLaunch` is going to be deleted from the private subnet.

File: guides/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,1138 @@
...
+    tag 'Name', "#{stack.cloud}-#{stack.name}-public"
+    vpc fn_ref('DemoVPC')
+    public_ips true
+  end
+end
+
+stack 'vpc', vpc
+```
+
+This adds a new property named "public_ips" to the `Mdoel::Template::Resource::EC2Subnet`
```
Comment (dgreene-r7) [inline guides/getting-started.md]: Spelling

File: guides/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,1138 @@
...
+      source '0.0.0.0/0'
+    end
+    egress_rule :tcp, 80 do
+      source '0.0.0.0/0'
+    end
+  end
+
+  ec2_instance 'NATInstance' do
+    tag 'Name', "#{stack.cloud}-#{stack.name}-nat"
+    image_id 'ami-c02b04a8'
```
Comment (dgreene-r7) [inline guides/getting-started.md:698]: How often does this AMI id change? Is someone going to have to update it every time Amazon releases a NAT AMI update?

**Comment by dgreene-r7** [comment]
This looks awesome!

`<pedant-mode>`
One quick tweak, instead of using the `text` code block I'd use `shell` or `bash` or something like that.
`</pedant-mode>`

**Comment by jmanero-r7** [comment]
Looks great! :+1:. Only comment is that you don't need to `require 'convection`' in Cloudfiles

**Comment by abegley-r7** [comment]
Excellent - just walked through the steps, no issues, all converged nicely!

**Comment by ghost** [comment]
This is wonderful!  Can we make this linked to from the readme.md?

Not sure how to find this or how to funnel people to it...

File: guides/getting-started.md (hunk 3 above)
Comment (fmitchell-r7) [inline guides/getting-started.md:698]: There are a lot of hard coded bits, like the AMI ID and the CIDR blocks. Touching on what @athompson-r7  mentioned, I feel like there's room for "Getting Started, Part 2" that walks through how to break a Cloudfile into maintainable parts and leverage attributes to create reusable templates.

Comment (ghost) [inline guides/getting-started.md:698]: At least with ami ids, is it worth putting something in that explicitly breaks with a clear cut error?  Is it worth adding something to the getting started guide to give people a one-liner to grab the latest id and explain where to insert it?

Is it worth adding the logic to the template to look it up for people and set the id?

I'm less concerned about the CIDR blocks than the hard coded ID's.

Additionally, I'm SURE the ID's change, just not sure how often they disappear.

Comment (fmitchell-r7) [inline guides/getting-started.md:698]: I updated the guide to link to [Amazon's docs on NAT instances](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_NAT_Instance.html) and described how to find the latest ID through the AWS web console. It looks like the NAT images update every six months or so. The oldest one is from two years ago.

**Comment by fmitchell-r7** [comment]
@eciramella-r7 I added a ink to the guide in the README. It's probably worth putting a link in [the code repo](https://github.com/rapid7/convection) back to this one.

**Pull request merged**